### PR TITLE
Add ppc64le support

### PR DESCRIPTION
**Pull request by drchristensen**
Unable to use GitVersion application on Linux systems with ppc64le architecture. For example: 
```
$ dotnet-gitversion /home/user/src/GitVersion
The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception.
```
Add support to build native ppc64le binaries along with other supported CPU architectures.

### CODE REVIEW

**Comment by bording** [comment]
I have questions about this change.

First, what sort of machine are you using that even has this sort of processor in it?

Second, where did you get a .NET distribution that works on it? Microsoft doesn't support or ship .NET for anything other than x64, arm32, and arm64.

**Comment by drchristensen** [comment]
IBM, Microsoft and RedHat have announced support for .NET 7/8 on POWER, you can find out more about it on the IBM .NET [blog](https://community.ibm.com/community/user/powerdeveloper/blogs/linda-alkire-kinnunen/2023/01/14/dotnet-on-ibm-power-resources-for-developers).  Microsoft supports installing .NET on Linux as described [here](https://learn.microsoft.com/en-us/dotnet/core/install/linux). In my use case, I'm building a project on Fedora 40 in a VM hosted on a P9 PowerNV system.  Installing the `GitVersion` tool is as simple as running `dotnet tool install --global GitVersion.Tool`.

**Comment by drchristensen** [comment]
The blog post also mentions support for IBM s390x mainframes, though I don't think the build containers used actually had the appropriate s390 cross compilers.

**Comment by bording** [comment]
Ok, so it's specifically Red Hat that is building a shipping .NET for that architecture.

Can you run `dotnet --info` and provide the output here? I want to confirm a few things (like what the RID actually shows up as) before merging this.

**Comment by bording** [comment]
Also, based on that, it doesn't seem like we'd need the `musl` variation since RHEL uses glibc and not musl.

File: Dockerfile.linux [hunk 1]
```diff
@@ -1,9 +1,10 @@
 FROM bording/crossbuild
 ARG ARCH='amd64'
-ENV CROSS_TRIPLE=${ARCH}
+ARG CC_ARCH='amd64'
```
Comment (bording) [inline Dockerfile.linux:3]: It's not clear to me why you've introduced this second ARCH variable.

I'm using QEMU already for the ARM builds, so I would think we'd be able to do the same thing for this new architecture in the same way.

**Comment by drchristensen** [comment]
Late reply on your `dotnet --info` question.  Output included below from Fedora 41: 
```
$ dotnet --info
.NET SDK:
 Version:           9.0.100-rc.2.24474.1
 Commit:            b4c156e3f6
 Workload version:  9.0.100-manifests.1d9add2b
 MSBuild version:   17.12.0-preview-24473-03+b4c156e3f

Runtime Environment:
 OS Name:     fedora
 OS Version:  41
 OS Platform: Linux
 RID:         fedora.41-ppc64le
 Base Path:   /usr/lib64/dotnet/sdk/9.0.100-rc.2.24474.1/

.NET workloads installed:
There are no installed workloads to display.
Configured to use loose manifests when installing new manifests.

Host:
  Version:      9.0.0-rc.2.24473.5
  Architecture: ppc64le
  Commit:       b4c156e3f6

.NET SDKs installed:
  8.0.110 [/usr/lib64/dotnet/sdk]
  9.0.100-rc.2.24474.1 [/usr/lib64/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.10 [/usr/lib64/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0-rc.2.24474.3 [/usr/lib64/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.10 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0-rc.2.24473.5 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/usr/lib64/dotnet]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Pull request closed without merging**